src/lib/tuya.ts

```diff
@@ -811,52 +811,54 @@ export const valueConverter = {
             return payload;
         },
     },
-    thermostatScheduleDayMultiDP: {
-        from: (v: string) => {
-            const schedule = [];
-            for (let index = 1; index < 17; index = index + 4) {
-                schedule.push(
-                    String(parseInt(v[index+0])).padStart(2, '0') + ':' +
-                    String(parseInt(v[index+1])).padStart(2, '0') + '/' +
-                    // @ts-ignore
-                    (parseFloat((v[index+2] << 8) + v[index+3]) / 10.0).toFixed(1),
-                );
-            }
-            return schedule.join(' ');
-        },
-        to: (v: string) => {
-            const payload = [0];
-            const transitions = v.split(' ');
-            if (transitions.length != 4) {
-                throw new Error('Invalid schedule: there should be 4 transitions');
-            }
-            for (const transition of transitions) {
-                const timeTemp = transition.split('/');
-                if (timeTemp.length != 2) {
-                    throw new Error('Invalid schedule: wrong transition format: ' + transition);
+    thermostatScheduleDayMultiDPN: (numberOfSchedules: number) => {
```

Comment:
What about setting default value for numberOfSchedules:

thermostatScheduleDayMultiDPN: (numberOfSchedules: number = 4)

With default value other thermostates definitions will not require update

Reply:
Good idea. I think there still is a small change in the other thermostats since we have to call method instead of using the field right? 

(Will need to apply this another, on mobile right now)

Reply:
Oh, indeed entry method for thermostats definition is:
thermostatScheduleDayMultiDPWithDayNumber, so it look like there also should be default value set.

After this failing tests should pass, but i think it is required to add additional tests to check new parameter anyway.

Reply:
Applied the default value as you suggested. I will also add you to my fork feel free to push any changes directly into the branch

Reply:
Applied the changes you suggested, I will also add you to my branch (it is your code anyway), feel free to push to the branch directly.